src/config.c

```diff
@@ -22,69 +22,68 @@
 
 #include "redisraft.h"
 
-const char *trace_names[] = {
-        "off",
-        "node",
-        "conn",
-        "raftlib",
-        "raftlog",
-        "generic",
-        "all"
+static const char *trace_names[] = {
+    "off",
+    "node",
+    "conn",
+    "raftlib",
+    "raftlog",
+    "generic",
+    "all"
 };
 
-int trace_flags[] = {
-        TRACE_OFF,
-        TRACE_NODE,
-        TRACE_CONN,
-        TRACE_RAFTLIB,
-        TRACE_RAFTLOG,
-        TRACE_GENERIC,
-        TRACE_ALL
+static const int trace_flags[] = {
+    TRACE_OFF,
+    TRACE_NODE,
+    TRACE_CONN,
+    TRACE_RAFTLIB,
+    TRACE_RAFTLOG,
+    TRACE_GENERIC,
+    TRACE_ALL
 };
+
 #define TRACE_FLAG_COUNT (sizeof(trace_flags) / sizeof(int))
 
-#define CONF_ID                         "id"
-#define CONF_ADDR                       "addr"
-#define CONF_PERIODIC_INTERVAL          "periodic-interval"
-#define CONF_REQUEST_TIMEOUT            "request-timeout"
-#define CONF_ELECTION_TIMEOUT           "election-timeout"
-#define CONF_CONNECTION_TIMEOUT         "connection-timeout"
-#define CONF_JOIN_TIMEOUT               "join-timeout"
-#define CONF_RESPONSE_TIMEOUT           "response-timeout"
-#define CONF_PROXY_RESPONSE_TIMEOUT     "proxy-response-timeout"
-#define CONF_RECONNECT_INTERVAL         "reconnect-interval"
-#define CONF_LOG_FILENAME               "log-filename"
-#define CONF_LOG_MAX_CACHE_SIZE         "log-max-cache-size"
-#define CONF_LOG_MAX_FILE_SIZE          "log-max-file-size"
-#define CONF_LOG_FSYNC                  "log-fsync"
-#define CONF_FOLLOWER_PROXY             "follower-proxy"
-#define CONF_QUORUM_READS               "quorum-reads"
-#define CONF_LOGLEVEL                   "loglevel"
-#define CONF_TRACE                      "trace"
-#define CONF_SHARDING                   "sharding"
-#define CONF_SLOT_CONFIG                "slot-config"
-#define CONF_SHARDGROUP_UPDATE_INTERVAL "shardgroup-update-interval"
-#define CONF_IGNORED_COMMANDS           "ignored-commands"
-#define CONF_EXTERNAL_SHARDING          "external-sharding"
-#define CONF_MAX_APPEND_REQ_IN_FLIGHT   "max-append-req-in-flight"
-#define CONF_TLS_ENABLED                "tls-enabled"
-#define CONF_CLUSTER_USER               "cluster-user"
-#define CONF_CLUSTER_PASSWORD           "cluster-password"
+static const char *conf_id                         = "id";
```

Comment:
Sorry, this is already a big PR, so, I couldn't stop to fix issues and pr got bigger. At least, it will address all config related changes (hopefully).